app/src/main/java/com/ably/tracking/demo/publisher/common/ListExtensions.kt

```diff
@@ -0,0 +1,4 @@
+package com.ably.tracking.demo.publisher.common
+
+fun <T> List<T>.copyAndReplaceElementAt(index: Int, element: T) =
+    subList(0, index) + element + subList(index, size - 1)
```

Comment:
I'm not entirely sure what this function should do, but it seems to put a new element at the given index and remove the last element :thinking: Is it intentional?
```kotlin
val list = listOf(1, 2, 3, 4, 5, 6)
// (1, 2, 3, 4, 5, 6)
   
list.copyAndReplaceElementAt(2, 9)
// (1, 2, 9, 3, 4, 5)
```

Reply:
nice catch 🙇 
0257ff398ec19440ba78d6237762081cadc6fffc

Reply:
Nice :+1: But now the name doesn't fit and I wonder what should be changed - the name or the implementation :thinking: Now it doesn't "copy and replace" but rather "copy and insert":
```kotlin
val list = listOf(1, 2, 3, 4, 5, 6)
// (1, 2, 3, 4, 5, 6)
   
list.copyAndReplaceElementAt(2, 9)
// (1, 2, 9, 3, 4, 5, 6)
```

Reply:
eb2fd2957f436f77d7630caa45c72521da70fa0a